experimental/sdata/timeseries/multi.go

```diff
@@ -0,0 +1,227 @@
+package timeseries
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/sdata"
+)
+
+// MultiFrame is a time series format where each series lives in its own single frame.
+// This time series format should be use for data that natively uses Labels and
+// when all of the series are not guaranteed to have identical time values.
+type MultiFrame []*data.Frame
+
+// NewMultiFrame creates an empty MultiFrame formatted time series.
+// This function must be called before the AddSeries Method.
+// The returned MultiFrame is a valid typed data response that corresponds to "No Data".
+func NewMultiFrame() *MultiFrame {
+	return &MultiFrame{
+		emptyFrameWithTypeMD(data.FrameTypeTimeSeriesMany),
+	}
+	// Consider: MultiFrame.New()
+}
+
+// values must be a numeric slice such as []int64, []float64, []*float64, etc or []bool / []*bool.
+func (mfs *MultiFrame) AddSeries(metricName string, l data.Labels, t []time.Time, values interface{}) error {
```

Comment:
here []time.Time should be []*time.Time, since we can miss data for certain data point, so the time would be able to be nil.